tools/cockpit.spec

```diff
@@ -20,6 +20,12 @@
 %define rhel 0
 %endif
 
+# for testing this already gets set in fedora.install, as we want the target
+# VERSION_ID, not the mock chroot's one
```

Comment:
Have you seen this anywhere else? What happens in the cockpit RPM is installed before rhel-release-server in the same transaction?

I'm worried about this. Does it make sense just to have RHEL 7.5 behavior here, and RHEL 7.4 behavior on the rhel-7-4 branch?

Reply:
What do you mean by "this"? Parsing os-release in the .spec is relatively common, e. g. [systemd's spec file](http://pkgs.fedoraproject.org/cgit/rpms/systemd.git/tree/systemd.spec) does that too. Our spec file also does it by default, but this doesn't work in our testsuite as that builds a RHEL package in a CentOS chroot.

I don't understand your question about install transactions -- this is purely a build-time thing, the built rpms have fixed dependencies, so OS install/upgrade orders don't matter here.

We want to be able to test current master on RHEL 7.4 (and 7.5), so we can't have the dependencies in the back-branches only. The conditional dependencies aren't new at all, they just get updated in this PR. The new thing is the more fine-grained differentiation between 7.4 and 7.5, but we have to do that *somehow*. Whether we do it directly in the spec or add some indirection in our build system that pre-processes the .spec.in and builds a target specific .spec doesn't matter that much, but IMHO the latter would be much more intrusive.

Reply:
> I don't understand your question about install transactions -- this is purely a build-time thing, the built rpms have fixed dependencies, so OS install/upgrade orders don't matter here.

Okay perfect. If build time only then I'm fine with this.